dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -10,22 +11,26 @@ class Api::V1::MlModelsController < Api::V1::JsonApiController
   # Save a trained ML model to S3 and a reference to it in the database.
   def save
     model_id = generate_id
-
-    return head :bad_request if params["ml_model"].nil? || params["ml_model"] == ""
-
-    metadata = params["ml_model"].except(:trainedModel, :featureNumberKey)
-    @user_ml_model = UserMlModel.create(
-      user_id: current_user&.id,
-      model_id: model_id,
-      name: params["ml_model"]["name"],
-      metadata: metadata.to_json
-    )
-    if @user_ml_model.persisted?
-      s3_filename = upload_to_s3(model_id, params["ml_model"].to_json)
-      status = s3_filename ? "success" : "failure"
-      render json: {id: model_id, status: status}
+    model_data = params["ml_model"]
+    return head :bad_request if model_data.nil? || model_data == ""
+    profanity_or_pii = ShareFiltering.find_failure(model_data.to_s, request.locale)
+    if profanity_or_pii
+      render json: {id: model_id, status: "failure", details: profanity_or_pii.type}
```

Comment:
Out of curiosity, what are the possible values of `profanity_or_pii.true`?  We want to make sure all possible values are unique to profanity filtering failures, so that the client code doesn't have to make too many assumptions about what type of error this is.  Otherwise, we could return a more explicit error indicating that it relates to profanity filtering...

Reply:
The possible values for `profanity_or_pii.type` are `email`, `phone`, `address` and `profanity` found [here](https://github.com/code-dot-org/code-dot-org/blob/977b0c3335fbe94d9b1392e0cb189bcaf5f9d07e/lib/cdo/share_filtering.rb#L14).

Reply:
I think I see what you mean, I updated `details` to be more specific.

Reply:
We decided that it wasn't important to specify the type of error so I modified to render a new status: `piiProfanity` that the code in AI Lab now handles to display the correct, generic message.